### ISSUE TITLE
.NET Core

### ISSUE THREAD

**Issue by ScorchedHands** [AvaloniaUI/Avalonia#314]
Will Perspex be supported on .NET Core?  We need a good GUI framework for .NET Core that will work cross-platform.  This looks like it would be a good WPF replacement for that use case.

**Comment by grokys** [comment]
Yes, we fully intend to get Perspex working with .NET core, but we're waiting for the final release; in particular the [.NET Standard Platform](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/standard-platform.md) will make things in general *so* much easier!

**Comment by mafshin** [comment]
I can't really wait to see .NET Core XAML apps running in Raspberry Pi 2

**Comment by umaranis** [comment]
Me too. I hope the project is making progress in this direction.

**Comment by grokys** [comment]
I think our main blocker will be that fact that according [to this](https://github.com/Reactive-Extensions/Rx.NET/issues/148) it appears that Rx isn't yet working on .NET core.

**Comment by mafshin** [comment]
So we are waiting till the chain of dependencies start supporting .NET Core.
I'm sure that will happen soon :+1:

**Comment by javagg** [comment]
This is great！

**Comment by ethanhs** [comment]
Well, with Core at 1.0 and that issue ~~closed~~ resolved, @grokys do you think work can start on this? I'd love to be able to use this with core. :)

**Comment by grokys** [comment]
Yes, it would be nice to try it out - please, give it a go!

**Comment by ethanhs** [comment]
>Yes, it would be nice to try it out - please, give it a go!
Sorry, I guess I wasn't clear. I won't be able to port it now (have my own project I have to work on atm). :/ but best of luck to whoever tries it. :)

I was more asking if there were any other blockers to porting.

**Comment by grokys** [comment]
Ah sorry, I thought you were wanting to do it. I'm not going to do it myself right now, as there's a lot to do on the base and I want to wait for the `.csproj` support as that will make it easier to port, I think. But if anyone else wants to give it a try, I'm all for it!

**Comment by mellinoe** [comment]
If the move to csproj is a blocker for you, do note that you can target .NET Core with MSBuild right now, it's just a little less "pretty" than it will be in the future. We have a document [here](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/target-dotnetcore-with-msbuild.md) describing the process. If anyone is interested in pursuing that route. I'd be more than willing to help out with getting the build projects set up correctly. Most of my own personal projects use MSBuild with .NET Core, so I'm fairly familiar with how it all works.

**Comment by grokys** [comment]
@mellinoe oh that is indeed interesting! I probably won't have time for the next few weeks, but someone else may well wish to try it.

**Comment by Seeker1437** [comment]
Ill give it a shot.

**Comment by grokys** [comment]
I've just had a quick go at converting our PCLs to .NET Standard and encountered the following problems:

- 'EmbeddedResource' doesn't seem to be supported
- Splat doesn't have .net standard version
- JetBrains.Annotations doesn't have .net standard version

**Comment by ScorchedHands** [comment]
You should be able to "import" the PCL platforms.  I know on project.json projects it looks something like:

  "frameworks": {
    "netstandard1.6": {
        "imports": "portable-net45+win8+dnxcore50"
    }
  }

I'm not sure how that translates to the csproj format though.

**Comment by grokys** [comment]
Thanks @ScorchedHands - I will try that! Here is the PR btw: https://github.com/AvaloniaUI/Avalonia/pull/620

**Comment by grokys** [comment]
@ScorchedHands that seems to have allowed me to import the splat nupkg but now I'm getting errors like:

```
Error	CS0518	Predefined type 'System.Boolean' is not defined or imported
```

Any idea what the correct fix to that is? I've looked around but can't find anything definite.

**Comment by ethanhs** [comment]
@grokys I think it has to do with the monikers you chose. Perhaps reading through [this blog post](https://blogs.msdn.microsoft.com/cesardelatorre/2016/06/28/running-net-core-apps-on-multiple-frameworks-and-what-the-target-framework-monikers-tfms-are-about/) will help a little?

**Comment by grokys** [comment]
@ethanhs that post is very useful in general, thanks! However I can't see anywhere it addresses the issue I'm seeing.

**Comment by ethanhs** [comment]
It doesn't specifically, I should have been more clear. Since it isn't importing the correct types, I believe you are missing a reference to eg dnxcore50. I am going to poke at this, just checked out your pr.

**Comment by ScorchedHands** [comment]
I think @ethanhs is correct and you're not importing quite the correct types.  Though instead of dnxcore50 you probably have to import net451 since the "unsupported" packages are probably based on the full framework.  This is overall the trickiest part of bridging the .net framework with .net core.

**Comment by SuperJMN** [comment]
OmniXAML v2 will support .NET Core!

**Comment by mellinoe** [comment]
For anyone interested in this, I rebased the PR I put up (#622) and was able to get the XamlTestApplication project to run on CoreCLR on Windows without any bugs (that I noticed :smile:). The PR isn't at a mergeable state yet, but the idea is there.

**Comment by feliwir** [comment]
Any news on this?

**Comment by grokys** [comment]
@feliwir not yet - I'm currently rewriting our renderer so it doesn't draw the whole window on every frame. I personally won't be looking into .net core until I've finished that.

**Comment by danwalmsley** [comment]
I know @curea is currently having a go.

**Comment by ronnyek** [comment]
Any updates? Id really like something to start building out some cross platform app and framework s

**Comment by kekekeks** [comment]
#875

**Comment by danwalmsley** [comment]
.net core support is now merged #875
